comp/core/config/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/DataDog/datadog-agent/comp/core/config
 
-go 1.21.0
+go 1.22.0
```

Comment:
Modules used by Otel have to stay at 1.21.0 so you can revert all changes to `go.mod` files